std/meta.d

```diff
@@ -1094,42 +1094,23 @@ template templateOr(Preds...)
     }
 }
 
+import std.traits : isIterable;
+
 /**
- * Converts an input range $(D range) to an alias sequence.
+ * Converts any foreach-iterable (e.g. an input range) $(D range) to an alias sequence.
  */
-template aliasSeqOf(alias range)
+template aliasSeqOf(alias iter)
+if (isIterable!(typeof(iter)))
 {
-    import std.traits : isArray, isNarrowString;
+    import std.traits : isNarrowString;
+    import std.array : array;
 
-    alias ArrT = typeof(range);
-    static if (isArray!ArrT && !isNarrowString!ArrT)
+    struct Impl
     {
-        static if (range.length == 0)
-        {
-            alias aliasSeqOf = AliasSeq!();
-        }
-        else static if (range.length == 1)
-        {
-            alias aliasSeqOf = AliasSeq!(range[0]);
-        }
-        else
-        {
-            alias aliasSeqOf = AliasSeq!(aliasSeqOf!(range[0 .. $/2]), aliasSeqOf!(range[$/2 .. $]));
-        }
-    }
-    else
-    {
-        import std.range.primitives : isInputRange;
-        static if (isInputRange!ArrT)
-        {
-            import std.array : array;
-            alias aliasSeqOf = aliasSeqOf!(array(range));
-        }
-        else
-        {
-            static assert(false, "Cannot transform range of type " ~ ArrT.stringof ~ " into a AliasSeq.");
-        }
+        static foreach (size_t i, el; iter.array)
```

Comment:
Solutions that didn't use `std.array.array` seemed more complicated than they were worth (which is probably not much, thinking of how current CTFE and newCTFE work).

Reply:
ping @UplinkCoder, I'd like to understand this better, because `static foreach` is just so very convenient.

Reply:
Don't be afraid too of using `static foreach`.
Entire Phobos uses it nowadays (-> https://github.com/dlang/phobos/pull/5989) and there was no measurable difference in the build time of the entire testsuite.

(Granted this doesn't say much about the extra-allocation of `.array`)